hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/LruBlockCache.java

```diff
@@ -481,7 +512,14 @@ private long updateSizeMetrics(LruCachedBlock cb, boolean evict) {
   @Override
   public Cacheable getBlock(BlockCacheKey cacheKey, boolean caching, boolean repeat,
       boolean updateCacheMetrics) {
-    LruCachedBlock cb = map.get(cacheKey);
+    LruCachedBlock cb = map.computeIfPresent(cacheKey, (key, val) -> {
```

Comment:
computeIfPresent is only atomic for ConcurrentHashMap, for other maps such as ConcurrentSkipListMap, the lambda function may be called multiple times. Is it safe here?

Reply:
Yeah, checked this before.  the map is ConcurrentHashMap, maybe we should define it as a ConcurrentHashMap explicitly. will address this in HBASE-22612.